daemons/fenced/fenced_history.c

```diff
@@ -330,6 +350,46 @@ stonith_local_history_diff(GHashTable *remote_history,
             }
     }
 
+    if (remote_history) {
+        init_stonith_remote_op_hash_table(&stonith_remote_op_list);
+
+        g_hash_table_iter_init(&iter, remote_history);
+        while (g_hash_table_iter_next(&iter, NULL, (void **)&op)) {
+
+            if (stonith__op_state_pending(op->state) &&
+                pcmk__str_eq(op->originator, stonith_our_uname, pcmk__str_casei)) {
+                crm_warn("Failing pending operation %.8s originated by us but "
+                         "known only from peer history", op->id);
+                op->state = st_failed;
+                op->completed = time(NULL);
+                /* use -EHOSTUNREACH to not introduce a new return-code that might
+                   trigger unexpected results at other places and to prevent
+                   remote_op_done from setting the delegate if not present
+                */
+                stonith_bcast_result_to_peers(op, -EHOSTUNREACH, FALSE);
+            }
+
+            updated = TRUE;
```

Comment:
We don't anywhere bail out of this loop - right?
So we could have something like
updates |= g_hash_table_size(remote_history);
outside the loop. Not just for efficiency as well
for the code to quickly show that we are updating
for any case where the table still has entries.

Reply:
Makes sense.